### PR TITLE
Migrate to User-Agent Client Hints blog post

### PR DESCRIPTION
**Pull request by rowan-m**
New article on UA-CH migration strategies

### CODE REVIEW

**Comment by rowan-m** [comment]
>   * Avoid [`please`](https://developers.google.com/style/word-list#please). Affected lines: 383

I think the `please` is in line with the styleguide here as it's asking people to raise an issue if the article hasn't given them what they need.

>   * Do not use [`crazy`](https://docs.google.com/spreadsheets/d/1v7RE9Buj_uFCP09uQnguvK-wilN8AVDx-t0B6J2biZk/edit#gid=1796212114&range=A23). Affected lines: 389

`crazy` is in the username of the Unsplash photo author, it's not part of the article.

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+use of the `User-Agent` HTTP header. You should also check your front-end code
+for use of already deprecated features, such as `navigator.platform` and
+`navigator.appVersion`.
+
+From a functional point of view, think about anywhere in your code where you are
+recording or processing:
+* Browser name or version
+* Operating system name or version
+* Device make or model
+* CPU type, architecture, or bitness (e.g. 64-bit)
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:52]:
```suggestion
* CPU type, architecture, or bitness (for example, 64-bit)
```

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+  // browser reports as compatible
+}
+```
+
+{% Aside 'gotchas' %} `userAgentData.brands` will contain varying values in a
+varying order, so don't rely on something appearing at a certain index. {%
+endAside %}
+
+If you need one of the more detailed, high-entropy user-agent values, you will
+need to specify it and check for the result in the returned Promise:
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:122]:
```suggestion
need to specify it and check for the result in the returned `Promise`:
```

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+surface](https://w3c.github.io/fingerprinting-guidance/#passive) in browsers, as
+well as being difficult to process. However, there are all kinds of valid
+reasons for collecting and processing user-agent data, so what's needed is a
+path to a better solution. User-Agent Client Hints provide both an explicit way
+to declare your need for user-agent data and methods to return the data in an
+easy-to-use format.
+
+{% Aside %} For more information on Client Hints and expanding them with
+user-agent data, read the [introductory article on User-Agent Client
+Hints](http://web.dev/user-agent-client-hints). {% endAside %}
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:30]:
```suggestion
Hints](/user-agent-client-hints). {% endAside %}
```

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+The resulting string would show the `Pixel 5` model, but still shows the reduced
+`92.0.0.0` as the `uaFullVersion` hint was not requested:
+
+```
+Mozilla/5.0 (Linux; Android 10.0; Pixel 5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.0.0 Mobile Safari/537.36
+```
+
+## Further support
+
+If these strategies do not cover your use case, please start a [Discussion in
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:383]:
```suggestion
If these strategies do not cover your use case, start a [Discussion in
```

**Comment by mihajlija** [comment]
Re: `please`, imo the sentence still sounds friendly enough when it's omitted because we suggest "exploring the issue together". Just a suggestion, it's a nit and we can go either way.

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+
+### Are you only using basic user-agent data?
+
+The default set of User-Agent Client Hints includes:
+* `Sec-CH-UA`: browser name and major/significant version
+* `Sec-CH-UA-Mobile`: boolean value indicating a mobile device
+* `Sec-CH-UA-Platform`: operating system name
+  * _Note:_ this has been updated in the spec and will be [reflected in
+    Chrome](https://groups.google.com/a/chromium.org/g/blink-dev/c/dafizBGwWMw/m/72l-1zm6AAAJ)
+    and other Chromium-based browsers shortly.
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:66]:
```suggestion
  * _Note that this has been updated in the spec and will be [reflected in
    Chrome](https://groups.google.com/a/chromium.org/g/blink-dev/c/dafizBGwWMw/m/72l-1zm6AAAJ)
    and other Chromium-based browsers shortly._
```

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+
+For example, let's say that `https://blog.site` hosts resources on
+`https://cdn.site` which can return resources optimized for a specific device.
+`https://blog.site` can ask for the `Sec-CH-UA-Model` hint, but needs to
+explicitly delegate it to `https://cdn.site` using the `Permissions-Policy`
+header. The list of policy-controlled hints is available in the [Clients Hints
+Infrastructure
+draft](https://wicg.github.io/client-hints-infrastructure/#policy-controlled-client-hints-features)
+
+⬇️ Response from blog.site delegating the hint
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:207]:
```suggestion
⬇️ Response from `blog.site` delegating the hint
```

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+
+```text
+Accept-CH: Sec-CH-UA-Model
+Permissions-Policy: ch-ua-model=(self "https://cdn.site")
+```
+
+⬆️ Request to subresources on cdn.site include the delegated hint
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:214]:
```suggestion
⬆️ Request to subresources on `cdn.site` include the delegated hint
```

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+
+```text
+Sec-CH-UA-Model: "Pixel 5"
+```
+
+You can specify multiple hints for multiple origins, and not just from the `ch-ua` range:
+
+⬇️ Response from blog.site delegating multiple hints to multiple origins
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:222]:
```suggestion
⬇️ Response from `blog.site` delegating multiple hints to multiple origins
```

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+{% Aside 'gotchas' %} You do **not** need to include each delegated hint in
+`Accept-CH`, but you **do** need to include `self` for each hint, even if you
+are not using it directly at the top-level. {% endAside %}
+
+## Strategy: Delegating hints to iframes
+
+Cross-origin iframes work in a similar way to cross-origin resources, but you
+specify the hints you would like to delegate in the `allow` attribute.
+
+⬇️ Response from blog.site
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:239]:
```suggestion
⬇️ Response from `blog.site`
```

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 10]
```diff
@@ -0,0 +1,391 @@
...
+
+```text
+Accept-CH: Sec-CH-UA-Model
+```
+
+↪️ HTML for blog.site
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:245]:
```suggestion
↪️ HTML for `blog.site`
```

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 11]
```diff
@@ -0,0 +1,391 @@
...
+
+```html
+<iframe src="https://widget.site" allow="ch-ua-model"></iframe>
+```
+
+⬆️ Request to widget.site
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:251]:
```suggestion
⬆️ Request to `widget.site`
```

File: src/site/content/en/blog/migrate-to-ua-ch/index.md [hunk 12]
```diff
@@ -0,0 +1,391 @@
+---
+layout: post
+title: Migrate to User-Agent Client Hints
+subhead: |
+  Strategies to migrate your site from relying on the user-agent string to the
+  new User-Agent Client Hints.
+authors:
+  - rowan_m 
+date: 2021-05-19
+description: |
```
Comment (mihajlija) [inline src/site/content/en/blog/migrate-to-ua-ch/index.md:10]:
```suggestion
description: |
  Strategies to migrate your site from relying on the user-agent string to the  new User-Agent Client Hints.
```

**Pull request merged**